specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-10-01-preview/cognitiveservices.json

```diff
@@ -11258,55 +11572,340 @@
       "additionalProperties": false
     },
     "AgentDeployment": {
-      "description": "Agent Deployment",
+      "description": "Type representing an agent deployment as a management construct.",
       "type": "object",
       "allOf": [
         {
           "$ref": "#/definitions/ResourceBase"
         }
       ],
       "properties": {
-        "deploymentId": {
-          "description": "Unique identifier for the deployment.",
+        "displayName": {
+          "description": "Gets or sets the display name of the deployment.",
           "type": "string",
           "x-nullable": true
         },
-        "agentId": {
-          "description": "Identifier of the agent being deployed.",
+        "id": {
+          "description": "Gets or sets the unique identifier of the deployment.",
           "type": "string",
           "x-nullable": true
         },
-        "environment": {
-          "description": "Deployment environment (e.g., development, staging, production).",
+        "associatedResourceId": {
+          "description": "The resource ID of the application associated with this deployment.",
           "type": "string",
           "x-nullable": true
         },
-        "status": {
-          "description": "Status of the deployment.",
+        "state": {
+          "description": "Gets or sets the current operational state of the deployment (and, intrinsically, of the comprising agents).",
+          "$ref": "#/definitions/AgentDeploymentState",
+          "x-nullable": true
+        },
+        "protocols": {
+          "description": "Gets or sets the supported protocol types and versions exposed by this deployment.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AgentProtocolVersion"
+          },
+          "x-nullable": true
+        },
+        "agents": {
+          "description": "Returns a flat list of agent:version deployed in this deployment.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VersionedAgentReference"
+          },
+          "x-nullable": true
+        },
+        "type": {
+          "description": "Gets or sets the type of deployment for the agent.",
+          "$ref": "#/definitions/AgentDeploymentType",
+          "x-nullable": true
+        },
+        "provisioningState": {
+          "description": "Gets or sets the provisioning state of the agent deployment.",
+          "$ref": "#/definitions/AgentDeploymentProvisioningState",
+          "readOnly": true
+        }
+      },
+      "additionalProperties": false
+    },
+    "AgentReference": {
+      "description": "Type modeling a reference to a version of an agent definition.",
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

[ARMBlockingComment] Don't reuse top-level property names within the property bag. This could be agentId (or something else you prefer), but would be better with a more specific name. Id is a very generic term: could mean name, number, guid, resourceId, etc.